### PR TITLE
Issue #SH-153 feat: Add any content to group implementation

### PR DESCRIPTION
**Pull request by sudip-mukherjee**

### CODE REVIEW

File: src/app/client/src/app/modules/groups/components/activity/activity-search/activity-search.component.ts [hunk 1]
```diff
@@ -239,9 +256,29 @@ export class ActivitySearchComponent implements OnInit, OnDestroy {
   }
 
   addActivity(event) {
-    const cdata = [{id: _.get(event, 'data.identifier'), type: 'Course'}];
+    const cdata = [{id: _.get(event, 'data.identifier'), type: _.get(event, 'data.contentType')}];
     this.addTelemetry('activity-course-card', cdata);
-    this.router.navigate(['/learn/course', _.get(event, 'data.identifier')], { queryParams: { groupId: _.get(this.groupData, 'id') } });
+
+    switch (_.get(event, 'data.contentType').toLowerCase()) {
```
Comment (vinukumar-vs) [inline src/app/client/src/app/modules/groups/components/activity/activity-search/activity-search.component.ts:262]: This should not be hardcoded like this.. There are only 3 types of details screen, 
1. For a collection of trackable:true => goto course details screen (course in this case)
2. For a collection of trackable=false => goto textbook details screen (Textbook)
3. For any non-collection content => goto resource details page (Resource, QuestionSet, etc..)

File: src/app/client/src/app/modules/groups/components/activity/add-activity-content-types/add-activity-content-types.component.ts [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    });
+    this.sendInteractData({id: `${cardData.activityType}-card`});
+    this.router.navigate([`${ADD_ACTIVITY_TO_GROUP}`, cardData.activityType , 1], { relativeTo: this.activatedRoute });
+  }
+
+  sendInteractData(interactData) {
+    const data = {
+      context: {
+        env: this.activatedRoute.snapshot.data.telemetry.env,
+        cdata: []
```
Comment (vinukumar-vs) [inline src/app/client/src/app/modules/groups/components/activity/add-activity-content-types/add-activity-content-types.component.ts:80]: CDATA should have group info, Activity type info

File: src/app/client/src/app/modules/groups/components/activity/add-activity-content-types/add-activity-content-types.component.ts [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+   }
+
+  ngOnInit() {
+    this.navigationHelperService.setNavigationUrl();
+    if (!this.csGroupAddableBloc.initialised) {
+      this.csGroupAddableBloc.init();
+    }
+    this.fetchActivityList();
+    this.activatedRoute.queryParams.subscribe((params) => {
+     this.groupData['name'] = _.get(params, 'groupName');
```
Comment (vinukumar-vs) [inline src/app/client/src/app/modules/groups/components/activity/add-activity-content-types/add-activity-content-types.component.ts:43]: Why are we doing this object mapping like this?

File: src/app/client/src/app/modules/groups/components/group-details/group-details.component.ts [hunk 4]
```diff
@@ -92,10 +92,15 @@ export class GroupDetailsComponent implements OnInit, OnDestroy {
...
-    this.toggleActivityModal(false);
-    this.addActivityModal.deny();
-    this.router.navigate([`${ADD_ACTIVITY_TO_GROUP}/${COURSES}`, 1], { relativeTo: this.activatedRoute });
+  navigateToAddActivity() {
+   this.router.navigate([`${ADD_ACTIVITY_CONTENT_TYPES}`], {
+     relativeTo: this.activatedRoute,
+     queryParams: {
+       groupName: _.get(this.groupData, 'name'),
+       createdBy: _.capitalize(_.get(_.find(this.groupData['members'], {userId: this.groupData['createdBy']}), 'name')),
+       groupId: _.get(this.groupData, 'id'),
```
Comment (vinukumar-vs) [inline src/app/client/src/app/modules/groups/components/group-details/group-details.component.ts:101]: This is the must parameter, Through error when this.groupData.id is not available. Don't navigate also.

File: src/app/client/src/app/modules/groups/interfaces/group.ts [hunk 5]
```diff
@@ -81,3 +81,9 @@ export interface IGroupMember {
   id?: string;
 }
 
+export interface IGroupData {
+  name: string;
```
Comment (vinukumar-vs) [inline src/app/client/src/app/modules/groups/interfaces/group.ts:85]: Why the creator & groupId names are difference here than what we get in activated route params?

File: src/app/client/src/app/modules/learn/components/course-consumption/course-consumption-header/course-consumption-header.component.html [hunk 6]
```diff
@@ -70,9 +70,10 @@ <h6 class="ui small info message" *ngIf="enrolledCourse && getBatchStatus() && b
 
       <div class="sb-g-col-xs-12 sb-g-col-md-3 sb-g-col-lg-3 sb-g-col-xxxl-4 certified-course">
 
-        <div class="certified-course__btn" *ngIf="showAddGroup">
+        <div class="certified-course__btn">
           <button class="sb-btn sb-btn-secondary sb-btn-normal ml-auto textbook__addbtn"
-            (click)="addActivityToGroup(); this.logTelemetry('add-to-group', courseHierarchy)">{{resourceService?.frmelmnts?.lbl?.AddtoGroup}}</button>
+          appAddToGroup [identifier]="courseHierarchy?.identifier" [pageId]="courseHierarchy?.contentType.toLowerCase()">
+          {{resourceService?.frmelmnts?.lbl?.AddtoGroup}}</button>
```
Comment (vinukumar-vs) [inline src/app/client/src/app/modules/learn/components/course-consumption/course-consumption-header/course-consumption-header.component.html:76]: Inform Ahment/Anil to change this "pageId" to "contextType"

File: src/app/client/src/app/modules/shared/directives/add-to-group/add-to-group.directive.ts [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+      this.toasterService.error(this.resourceService.messages.emsg.noAdminRole);
+    }
+  }
+
+  sendInteractData() {
+    const data = {
+      context: {
+        env: 'groups',
+        cdata: [{
+          type: _.get(this.groupAddableBlocData, 'params.contentType'),
```
Comment (vinukumar-vs) [inline src/app/client/src/app/modules/shared/directives/add-to-group/add-to-group.directive.ts:86]: This should be always capitalize word. like "Course"/"Textbook"

File: src/app/helpers/whitelistApis.js [hunk 8]
```diff
@@ -551,6 +551,10 @@ const API_LIST = {
     '/learner/anonymous/otp/v1/generate': {
       checksNeeded: []
     },
+    '/learner/data/v1/form/read': {
```
Comment (vinukumar-vs) [inline src/app/helpers/whitelistApis.js:554]: Form/Read should be already existing. This is not the new end-point we are using first time.

Comment (vinukumar-vs) [reply src/app/helpers/whitelistApis.js:554]: Not required

**Review by vinukumar-vs — CHANGES_REQUESTED** [review]

File: src/app/client/src/app/modules/groups/components/activity/activity-search/activity-search.component.ts (hunk 1 above)
Comment (sudip-mukherjee) [reply src/app/client/src/app/modules/groups/components/activity/activity-search/activity-search.component.ts:262]: how trackable is coming into this scenario ? we haven't had a discussion regarding trackable in this redirection ?

File: src/app/client/src/app/modules/groups/components/group-details/group-details.component.ts (hunk 4 above)
Comment (sudip-mukherjee) [reply src/app/client/src/app/modules/groups/components/group-details/group-details.component.ts:101]: not required. No need to pass groupId in the query params, I have changed the code. while setting up the pageContext, groupId is must,

File: src/app/client/src/app/modules/groups/interfaces/group.ts (hunk 5 above)
Comment (sudip-mukherjee) [reply src/app/client/src/app/modules/groups/interfaces/group.ts:85]: not required. changed the code.

File: src/app/client/src/app/modules/groups/components/activity/add-activity-content-types/add-activity-content-types.component.ts (hunk 3 above)
Comment (sudip-mukherjee) [reply src/app/client/src/app/modules/groups/components/activity/add-activity-content-types/add-activity-content-types.component.ts:43]: modified the code.

File: src/app/client/src/app/modules/learn/components/course-consumption/course-consumption-header/course-consumption-header.component.html (hunk 6 above)
Comment (sudip-mukherjee) [reply src/app/client/src/app/modules/learn/components/course-consumption/course-consumption-header/course-consumption-header.component.html:76]: sure

File: src/app/helpers/whitelistApis.js (hunk 8 above)
Comment (sudip-mukherjee) [reply src/app/helpers/whitelistApis.js:554]: I checked that if this '/learner/data/v1/form/read' is not added in the  white list APIs, then form API call which is going from csl side to get the supported activity list is not working, giving 403

File: src/app/client/src/app/modules/groups/components/activity/add-activity-content-types/add-activity-content-types.component.ts (hunk 2 above)
Comment (sudip-mukherjee) [reply src/app/client/src/app/modules/groups/components/activity/add-activity-content-types/add-activity-content-types.component.ts:80]: done

**Review by vinukumar-vs — APPROVED** [review]

**Pull request merged**